packages/react-router/lib/dom/ssr/single-fetch.tsx

```diff
@@ -32,15 +31,22 @@ export type SingleFetchRedirectResult = {
   replace: boolean;
 };
 
+// Shared/serializable type used by both turbo-stream and RSC implementations
+type DecodedSingleFetchResults =
+  | { routes: { [key: string]: SingleFetchResult } }
+  | { redirect: SingleFetchRedirectResult };
```

Comment:
`fetchAndDecode` will now return this shape which can be the common payload for turbo-stream and RSC.  `unwrapSingleFetchResult` now operates on this so that implementation can be shared.